test/stdlib/MetadataPruning.swift

```diff
@@ -0,0 +1,133 @@
+// RUN: %target-run-simple-swift
+// RUN: %target-run-simple-swift(-Xfrontend -disable-reflection-metadata)
+// RUN: %target-run-simple-swift(-Xfrontend -reflection-metadata-for-debugger-only)
+// RUN: %target-run-simple-swift(-Xfrontend -disable-reflection-names)
+//
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+var tests = TestSuite("MetadataPruning")
+
+struct TestStruct {
```

Comment:
Most of these test classes are copied from `ForEachField.swift` test.

Reply:
Good idea, this gives us nice coverage.